src/components/banner/index.tsx

```diff
@@ -82,93 +84,127 @@ export const Banner = (props: EmbeddedMessageData) => {
-            >
-              {message?.elements?.body}
-            </text>
-          </div>
-          {message?.elements?.mediaUrl && (
-            <img
-              style={{ ...defaultImageStyles, ...imgStyle }}
-              src={message?.elements?.mediaUrl}
-            />
-          )}
-        </div>
-        <div style={bannerButtons}>
-          {message?.elements?.buttons?.map((button: any, index: number) => (
-            <button
-              key={index}
-              disabled={index === 0 ? disablePrimaryBtn : disableSecondaryBtn}
-              style={
-                index === 0
-                  ? disablePrimaryBtn
-                    ? { ...defaultButtonStyles, ...primaryDisableBtnStyle }
-                    : { ...defaultButtonStyles, ...primaryBtnStyle }
-                  : disableSecondaryBtn
-                  ? { ...defaultButtonStyles, ...secondaryDisableBtnStyle }
-                  : { ...defaultButtonStyles, ...secondaryBtnStyle }
-              }
-              onClick={() => {
-                const clickedUrl =
-                  button?.action?.data?.trim() || button?.action?.type || '';
-                embeddedManager.handleEmbeddedClick(
-                  message,
-                  button?.id,
-                  clickedUrl
-                );
-                embeddedManager.trackEmbeddedClick(
-                  message,
-                  button?.id,
-                  clickedUrl
-                );
-              }}
-            >
-              {button.title ? button.title : `Button ${index + 1}`}
-            </button>
-          ))}
+  const handleButtonClick = (button: any) => {
+    const clickedUrl =
+      button?.action?.data?.trim() || button?.action?.type || '';
+    embeddedManager.handleEmbeddedClick(message, button?.id, clickedUrl);
+    embeddedManager.trackEmbeddedClick(message, button?.id, clickedUrl);
+  };
+
+  function addButtonClickEvent(button: HTMLElement, index: number) {
+    button.addEventListener('click', (event) => {
+      // Prevent the click event from bubbling up to the div
+      event.stopPropagation();
+      if (!message?.elements?.buttons) {
+        return '';
+      }
+      handleButtonClick(message?.elements?.buttons[index]);
+    });
+  }
+
+  setTimeout(() => {
+    const bannerDiv = document.getElementsByName(
+      `${message?.metadata?.messageId}-banner`
+    )[0];
+    const primaryButtonClick = document.getElementsByName(
+      `${message?.metadata?.messageId}-banner-primaryButton`
+    )[0];
+    const secondaryButtonClick = document.getElementsByName(
+      `${message?.metadata?.messageId}-banner-secondaryButton`
+    )[0];
+    if (bannerDiv) {
+      bannerDiv.addEventListener('click', handleBannerClick);
+    }
+    if (primaryButtonClick) {
+      addButtonClickEvent(primaryButtonClick, 0);
+    }
+    if (secondaryButtonClick) {
+      addButtonClickEvent(secondaryButtonClick, 1);
+    }
+  }, 0);
```

Comment:
just curious why this is a timeout?

Reply:
There is a concept in js called Event loop. It manages asynchronous operations. The event loop works by constantly monitoring two queues: the call stack and the callback queue. When the event loop starts, it checks the call stack. If the call stack is empty, the event loop will take the next function from the callback queue and move it to the call stack. The function will then be executed.

In this code, setTimeout schedules the execution of the block after the current stack clears, allowing time for DOM elements to render. Without setTimeout, the code executes synchronously, attempting to access DOM elements before they exist, leading to undefined results.
You can learn about event loop with example here: https://www.jsv9000.app/

Reply:
Got it, it's just there to handle the case where the dom isn't rendered yet. In that case, we should allow our customer's to handle the hook up of the event handling. 

Meaning that the sample app should do this attachment but it shouldn't be done in this component. 

cc: @bradumbaugh so you are aware and can provide input as necessary.

Reply:
Hi @mprew97 I am NOT sure what you mean but we need to handle these click events inside the components because inside click events of the buttons and the main view we handle clicked url and also the tracking of embedded click.

Reply:
hmmm, i don't think `setTimeout` is necessary still. You should theoretically just be able to create a function that gets called after the component is rendered. `setTimeout` seems like overkill https://teamtreehouse.com/community/how-do-i-set-an-event-listener-on-an-element-that-doesnt-yet-exist

Reply:
Hi @mprew97 We have been trying alternative approaches, but unfortunately, we are still facing multiple click issues with other approaches. Regarding the setTimeout function, we won't have any network speed-related problems as we are rendering components based on message data that has no dependency on the network. Indeed, setTimeout is not the ideal solution, so we will try for a more appropriate one.